src/openeo_gfmap/utils/catalogue.py

```diff
@@ -240,9 +264,15 @@ def select_s1_orbitstate_vvvh(
     backend: BackendContext,
     spatial_extent: SpatialContext,
     temporal_extent: TemporalContext,
+    max_temporal_gap: int = 30,
```

Comment:
I am wondering if this indeed needs to be a preset value or whether it can be made optional?

From a thematic context, is this always a requirement?

Reply:
that's good to know. Will the standard preset of 30 capture all possible observations around the globe, or might it be longer if you go to areas with less coverage?

the reason I am asking is because we do need to make sure that this default setting will not impede any external users in the long-run

Reply:
Before the outage of S1B, the worldwide minimum coverage would be at least once every 12 days. With S1B out, there's now some regions having only once every 24 days and some even without any. It's hard to set a universal threshold. But we could surely relax it a bit. Alternatively we could separate between a threshold to prefer one orbit over the other, and another one that completely fails the catalogue query because of a gap too wide. The accumulated are overlap in principle looks at observation frequency, but just doesn't take into account long gaps. Remembering from HRL crop types, I do think that indeed the 30 may be a bit too strict.

Reply:
How long would you set it up then @kvantricht ? 60 days ?

Reply:
OK so I checked this again, and in fact the confusing part is that `max_temporal_gap` threshold is never used to fail the orbit selection step. It's only used to promote one over the other if one has gap smaller than this threshold and the other does not. It looks now by the way that if none of the orbits fully covers the bbox and if one has a gap smaller than the threshold and the other larger, that this orbit will be chosen, even though it doesn't fully cover the bbox.

So maybe the check should still be changed, so that:
1. `if not ascending_overlap and not descending_overlap: raise UncoveredS1Exception()`
2. If one overlaps and the other does not: choose this one
3. If both overlap, check if one has a temporal gap that is too wide: if so, choose the other one
4. If both overlap and either none (or both!) exceed the temporal gap: check the cumulated area intersection

Reply:
@HansVRP Does it seems good for you? Should I resolve this conversation?

Reply:
I think this check is indeed sensible. If the code was changed accordingly we can resolve the check

Reply:
Added